data/ai/lua/generic_recruit_engine.lua

```diff
@@ -21,6 +21,8 @@ return {
     --   finding the best unit types to recruit. It does not work otherwise. It assumes normal vision of the AI side (that is, it disregards
     --   hidden enemy units) for determining from which keep hex the leader should recruit and on which castle hexes to recruit new units
     init = function(ai_cas, params)
+        wesnoth.deprecated_message('ai/lua/generic_recruit_engine.lua', 3, '1.19', 'All its functionality has been moved to ai/lua/ca_recruit_rushers.lua which can now be used directly.')
```

Comment:
I think I'd just move this to the top of the file, above `return`.

Reply:
Changed and force-pushed.